### PR TITLE
Move contrib from abibuilder to GitHub

### PR DESCRIPTION
**Pull request by t0mdavid-m**
## Description

This PR changes the `openms-ci-full` CI job to use the contrib build files from the [github repo](https://github.com/OpenMS/contrib) instead of the abibuilder.
## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

## **Type**
enhancement

___

## **Description**
- Updated the CI configuration to dynamically fetch the latest contrib build for Windows from a GitHub repository using the GitHub API, enhancing the maintainability and flexibility of the CI process.
- Added `GITHUB_TOKEN` as an environment variable to authenticate requests to the GitHub API.
- Modified the AUTHORS file to include a new contributor, Tom David Müller, acknowledging their contributions to the project.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>openms_ci_matrix_full.yml</strong><dd><code>Update CI to Fetch Contrib Builds from GitHub Instead of Static URL</code></dd></summary>
<hr>
      
.github/workflows/openms_ci_matrix_full.yml

<li>Changed the source of the contrib build files from a static URL to <br>dynamically fetching the latest release from a GitHub repository using <br>GitHub API.<br> <li> Added environment variable <code>GITHUB_TOKEN</code> to authenticate GitHub API <br>requests.<br> <li> Replaced the static download URL with a script to fetch and download <br>the latest <code>contrib_build-Windows.tar.gz</code> from the specified GitHub <br>repository.<br> <li> Added commands to extract the downloaded tar.gz file.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7482/files#diff-0be71f2159fab4dd562ae195019726d8ef93f807d2e9b91b67494e57c1419f1c">+9/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>AUTHORS</strong><dd><code>Update Authors List to Include New Contributor</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
AUTHORS

- Added Tom David Müller to the list of authors.

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7482/files#diff-ab6af77435f58cc0c9d4c31dfe05656e45187cc7c7fc02aada401a7642125463">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .github/workflows/openms_ci_matrix_full.yml [hunk 1]
```diff
@@ -288,9 +288,17 @@ jobs:
...
       if: startsWith(matrix.os, 'windows') && steps.cache-contrib.outputs.cache-hit != 'true'
       shell: bash
+      env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       run: |
           cd OpenMS/contrib
-          curl --no-progress-meter -L -o contribbld.tar.gz https://abibuilder.cs.uni-tuebingen.de/archive/openms/contrib/${{ steps.set-vars.outputs.contrib_os }}/${{ steps.set-vars.outputs.contrib_os_ver }}x64/${{ steps.set-vars.outputs.contrib_compiler }}contrib_build.tar.gz
+          # Use GitHub API to get the download URL for the latest release asset named 'contrib_build-Windows.tar.gz'
+          DOWNLOAD_URL=$(curl -sH "Authorization: token $GITHUB_TOKEN" \
+            https://api.github.com/repos/t0mdavid-m/contrib/releases/latest \
```
Comment (timosachsenberg) [inline .github/workflows/openms_ci_matrix_full.yml]: use OpenMS org repo

**Review by timosachsenberg — CHANGES_REQUESTED** [review]
thanks! see comment

File: .github/workflows/openms_ci_matrix_full.yml (hunk 1 above)
Comment (t0mdavid-m) [reply .github/workflows/openms_ci_matrix_full.yml]: Thank you! I completely forgot to change that.

**Comment by jpfeuffer** [comment]
Nice! 
I think these days one could just use
`gh release download -R MyOrg/MyRepo --pattern 'myproject-*.tar.gz'`

**Comment by jpfeuffer** [comment]
Oh yes you need an

```
  env:
    GH_TOKEN: ${{ github.token }}
```

for the download step then. Has the same meaning as passing it to curl before.

**Comment by t0mdavid-m** [comment]
Thanks a lot for the feedback. I did not know this was possible via the GitHub CLI.

If there is no further feedback, the PR would be ready to merge from my side.

**Review by timosachsenberg — APPROVED** [review]
thanks!

**Pull request merged**